deploy/eck-elasticsearch/values.yaml

```diff
@@ -85,11 +137,208 @@ nodeSets:
     #
     node.store.allow_mmap: false
   podTemplate:
+    # The following spec is exactly the Kubernetes Core V1 PodTemplateSpec. Any fields within the PodTemplateSpec
+    # are supported within the 'spec' field below.  Please see below documentation for the exhaustive list of fields.
+    #
+    # https://v1-24.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#podtemplatespec-v1-core
+    #
+    # Only the commonly overridden/used fields will be noted below.
+    # 
     spec:
+
+    # If specified, the pod's scheduling constraints
+    # https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-advanced-node-scheduling.html
+    # affinity:
+    #   nodeAffinity:
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #       nodeSelectorTerms:
+    #       - matchExpressions:
+    #         - key: topology.kubernetes.io/zone
+    #           operator: In
+    #           values:
+    #           - antarctica-east1
+    #           - antarctica-west1
+    #     preferredDuringSchedulingIgnoredDuringExecution:
+    #     - weight: 1
+    #       preference:
+    #         matchExpressions:
+    #         - key: another-node-label-key
+    #           operator: In
+    #           values:
+    #           - another-node-label-value
+    #   podAffinity:
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #     - labelSelector:
+    #         matchExpressions:
+    #         - key: security
+    #           operator: In
+    #           values:
+    #           - S1
+    #       topologyKey: topology.kubernetes.io/zone
+    #   podAntiAffinity:
+    #     preferredDuringSchedulingIgnoredDuringExecution:
+    #     - weight: 100
+    #       podAffinityTerm:
+    #         labelSelector:
+    #           matchLabels:
+    #             elasticsearch.k8s.elastic.co/cluster-name: quickstart
+    #         topologyKey: kubernetes.io/hostname
```

Comment:
I would probably shorten this to use just one example, shorter examples are easier to read and understand.